decorator.py

```diff
@@ -0,0 +1,31 @@
+def type_check(correct_type):
+    """
+    decorator function, checks if the variable is of the required type
+    :param correct_type: required type of variable
+    :return: fun with var or a message if the variable is not of the appropriate type
+    """
+    def decorator(fun):
```

Comment:
replace 'fun' Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing

Reply:
I changed it to func